### PR TITLE
Validations, more form styles, tests

### PR DESCRIPTION
**Pull request by Ayerbe88**

### CODE REVIEW

File: webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.spec.ts [hunk 1]
```diff
@@ -36,5 +41,27 @@ describe('FormFieldControlService', () => {
     const formFields = [formField1, formField2, formField3, formField4];
     const formGroup = service.toFormGroup(formFields);
     expect(Object.keys(formGroup.value).length).toEqual(2);
+    expect(formGroup).toBeInstanceOf(FormGroup);
+  });
+
+  it('should create validation messages', () => {
```
Comment (stefanleminh) [inline webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.spec.ts:47]: Maybe give another detail in the title? Such as, when/under which circumstance the validation message is supposed to appear

File: webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.ts [hunk 2]
```diff
@@ -1,20 +1,74 @@
...
 import {FormFieldBase} from './form-field-base';
 
 @Injectable()
 export class FormFieldControlService {
   constructor() {}
 
+  private emailPatternRegex: string =
+    '^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$';
+  private zipCodePatternRegex: string = '[0-9]{4}';
+  private numberPatternRegex: string = '[0-9]';
```
Comment (stefanleminh) [inline webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.ts:14]: Is this correct? Only a single digit for number fields would mean we cant input numbers for the price or anything above 9.

File: webcomponents/search-webcomponent/src/assets/i18n/de.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
-  "home": {
-    "categories-description": "Suchen Sie relevante Informationen",
-    "search-form": "Suche"
-  }
+  "categories-description": "Suchen Sie relevante Informationen",
+  "search-form": "Suche",
+  "offers": "Offers",
```
Comment (stefanleminh) [inline webcomponents/search-webcomponent/src/assets/i18n/de.json:4]: Needs a German translation

File: webcomponents/search-webcomponent/src/assets/i18n/de.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
-  "home": {
-    "categories-description": "Suchen Sie relevante Informationen",
-    "search-form": "Suche"
-  }
+  "categories-description": "Suchen Sie relevante Informationen",
+  "search-form": "Suche",
+  "offers": "Offers",
+  "searches": "Searches",
```
Comment (stefanleminh) [inline webcomponents/search-webcomponent/src/assets/i18n/de.json:5]: Needs a German translation

**Review by stefanleminh — CHANGES_REQUESTED** [review]
Looks good! Just a few small things that should be adjusted and we should be good to merge this.

File: webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.ts (hunk 2 above)
Comment (Ayerbe88) [reply webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.ts:14]: You are right that's wrong, changed it to '^((?![0-9]).)*$'

Comment (stefanleminh) [reply webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.ts]: Wouldn't it be more something like `^([0-9])+$` ? The one you wrote matches any character *except* for numbers between 0 and infinite times

Comment (stefanleminh) [reply webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.ts]: Okay I think you coded the right thing, just the one in the comment isn't the right one

**Review by stefanleminh — APPROVED** [review]
Looks good now!

File: webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.ts (hunk 2 above)
Comment (Ayerbe88) [reply webcomponents/search-webcomponent/src/app/shared/form/form-field-control.service.ts]: :S yea sorry was testing the other as well and copy pasted wrong...

**Pull request merged**